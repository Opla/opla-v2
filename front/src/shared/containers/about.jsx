/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { appSetTitleName } from "zoapp-front/dist/actions/app";

const aboutStyle = {
  margin: "5% 40% 10% 10%",
};

class About extends Component {
  constructor(props) {
    super();
    props.appSetTitleName("About");
  }

  render() {
    const { isLoading } = this.props;

    return (
      <div style={aboutStyle}>
        {isLoading ? "" : null}
        <h1>Essential</h1>
        <h4>Your own virtual assistant</h4>
        <p>
          We are Opla, the mighty volcanic team! Since 2015 we started
          developing this deeptech chatbot solution. Nowadays we are truly
          commited to OpenSource and fully involved in creating the most
          complete and user oriented conversational interface tool.
        </p>
        <h1>Version</h1>
        <h4>V0.1 - Amazing Alou </h4>
        <p>
          (Le Puy Alou : a french volcano) July 2018<br />
          <a
            href="https://github.com/Opla/"
            target="_blank"
            rel="noreferrer noopener"
          >
            See more on Github
          </a>
        </p>
        <h1>Contributors</h1>
        <h4>Thanks to the team and contributors</h4>
        <p>
          Elodie, Julie, Maali, Marie, Nicolle, Adil, Dorian, Florian, Loic,
          Manu, Mik, Thomas, Will, Will<br />
          <a
            href="https://github.com/Opla/joinUs"
            target="_blank"
            rel="noreferrer noopener"
          >
            And we are hiring !
          </a>
        </p>
        <p>
          Follow us on Twitter
          <a
            href="https://twitter.com/opla_ai/"
            target="_blank"
            rel="noreferrer noopener"
          >
            @opla_ai
          </a>
          <br />
          Or contact us hello@opla.ai
        </p>
      </div>
    );
  }
}

About.propTypes = {
  isLoading: PropTypes.bool,
  appSetTitleName: PropTypes.func.isRequired,
};
const mapDispatchToProps = (dispatch) => ({
  appSetTitleName: (titleName) => {
    dispatch(appSetTitleName(titleName));
  },
});

// prettier-ignore
export default connect(
  null,
  mapDispatchToProps,
)(About);
