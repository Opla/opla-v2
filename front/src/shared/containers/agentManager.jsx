/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Grid, Inner, Cell } from "zrmc";
import Loading from "zoapp-front/components/loading";
import SignInForm from "zoapp-front/containers/signInForm";

import ExplorerContainer from "./explorerContainer";
import IntentContainer from "./builder/intentContainer";
import EntityContainer from "./builder/entityContainer";
import FunctionContainer from "./builder/functionContainer";

class AgentManager extends Component {
  render() {
    let { isLoading } = this.props;
    if (!isLoading && !this.props.intents && this.props.isSignedIn) {
      isLoading = true;
    }
    if (!this.props.isSignedIn) {
      return <SignInForm />;
    } else if (this.props.intents == null) {
      return <Loading />;
    }
    let panel1 = null;
    let panel2 = null;

    panel1 = (
      <Cell
        style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
        className="zui-color--white zui-panel"
        span={3}
      >
        <ExplorerContainer
          onRename={this.props.onRenameIntent}
          onAdd={this.props.onAddIntent}
          onDelete={this.props.onDeleteIntent}
        />
      </Cell>
    );
    if (this.props.selectedType === "function") {
      panel2 = (
        <Cell
          style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
          className="zui-color--white zui-panel"
          span={9}
        >
          <FunctionContainer />
        </Cell>
      );
    } else if (this.props.selectedType === "entity") {
      panel2 = (
        <Cell
          style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
          className="zui-color--white zui-panel"
          span={9}
        >
          <EntityContainer />
        </Cell>
      );
    } else {
      panel2 = (
        <Cell
          style={{ margin: "0px", backgroundColor: "#f2f2f2" }}
          className="zui-color--white zui-panel"
          span={9}
        >
          <IntentContainer handleRename={this.handleRenameIntent} />
        </Cell>
      );
    }
    return (
      <Grid
        gutter={{ desktop: "0px", tablet: "0px", phone: "0px" }}
        style={{ margin: "0px", padding: "0px" }}
      >
        <Inner style={{ gridGap: "0px" }}>
          {panel1}
          {panel2}
        </Inner>
      </Grid>
    );
  }
}

AgentManager.defaultProps = {
  bot: null,
  intents: null,
  selectedIntent: null,
  selectedBotId: null,
  store: null,
  activeTab: 0,
};

AgentManager.propTypes = {
  activeTab: PropTypes.number,
  isLoading: PropTypes.bool.isRequired,
  isSignedIn: PropTypes.bool.isRequired,
  selectedBotId: PropTypes.string,
  selectedType: PropTypes.string.isRequired,
  bot: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
    language: PropTypes.string,
  }),
  intents: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string })),
  selectedIntentIndex: PropTypes.number.isRequired,
  selectedIntent: PropTypes.shape({ id: PropTypes.string }),
  store: PropTypes.shape({}),
  titleName: PropTypes.string.isRequired,

  onRenameIntent: PropTypes.func.isRequired,
  onAddIntent: PropTypes.func.isRequired,
  onDeleteIntent: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  let { selectedIntent } = state.app;
  const selectedBotId = state.app ? state.app.selectedBotId : null;
  // TODO get selectedBot from selectBotId
  const bot = selectedBotId ? admin.bots[0] : null;
  const intents = state.app.intents ? state.app.intents : null;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.loading || false;
  const selectedIntentIndex = state.app ? state.app.selectedIntentIndex : 0;
  const selectedType = state.app ? state.app.selectedType : "intent";
  if (!selectedIntent) {
    selectedIntent = state.app.intents
      ? state.app.intents[selectedIntentIndex]
      : null;
  }
  const titleName = state.app.activeScreen.name
    ? state.app.activeScreen.name
    : "";
  return {
    selectedBotId,
    bot,
    intents,
    isLoading,
    isSignedIn,
    selectedIntent,
    selectedIntentIndex,
    selectedType,
    titleName,
  };
};

const mapDispatchToProps = () => ({});

// prettier-ignore
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AgentManager);
