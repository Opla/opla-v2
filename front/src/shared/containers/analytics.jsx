/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Grid, Inner, Cell } from "zrmc";
import Loading from "zoapp-front/components/loading";
import Panel from "zoapp-front/components/panel";
import { apiGetMetricsRequest } from "../actions/api";

export class AnalyticsBase extends Component {
  componentDidMount() {
    if (this.props.selectedBotId) {
      this.props.fetchMetrics(this.props.selectedBotId);
    }
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.selectedBotId &&
      this.props.selectedBotId !== prevProps.selectedBotId
    ) {
      this.props.fetchMetrics(this.props.selectedBotId);
    }
  }

  /* eslint-disable class-methods-use-this */
  renderingValue(value, unit) {
    let res;
    if (Number.isNaN(Number(value)) || value === null) {
      res = <span className="opla-analytics_error">No data</span>;
    } else {
      res = (
        <span className="opla-analytics_value">
          {value}
          <span className="opla-analytics_unit">{unit}</span>
        </span>
      );
    }
    return res;
  }

  render() {
    const { isLoading, isSignedIn, metrics } = this.props;

    if (!isSignedIn) {
      return <div>You need to sign in...</div>;
    }

    if (isLoading || !metrics) {
      return <Loading />;
    }

    return (
      <div className="opla-analytics zui-layout__content zui-color--grey-100">
        <Grid>
          <Inner>
            <Cell className="zui-color--white" span={12}>
              <Panel title="Key metrics">
                <div className="opla-analytics_subpanel">
                  <div className="opla-analytics_metric">
                    <div>
                      {this.renderingValue(metrics.users.count)}
                      <div className="opla-analytics_legend">Users</div>
                    </div>
                  </div>
                  <div className="opla-analytics_metric">
                    <div>
                      {this.renderingValue(metrics.conversations.count)}
                      <div className="opla-analytics_legend">Conversations</div>
                    </div>
                  </div>
                  <div className="opla-analytics_metric">
                    <div>
                      {this.renderingValue(
                        metrics.conversations.messages_per_conversation,
                      )}
                      <div className="opla-analytics_legend">Messages</div>
                    </div>
                  </div>
                </div>
              </Panel>
            </Cell>
          </Inner>
          <Inner>
            <Cell
              className="zui-color--white"
              style={{ paddingTop: "96px" }}
              span={12}
            >
              <Panel title="Platform metrics">
                <div className="opla-analytics_subpanel">
                  <div className="opla-analytics_metric">
                    <div>
                      {this.renderingValue(metrics.sessions.duration, "ms")}
                      <div className="opla-analytics_legend">
                        Session duration
                      </div>
                    </div>
                  </div>
                  <div className="opla-analytics_metric">
                    <div>
                      {this.renderingValue(
                        (metrics.errors.rate * 100).toFixed(2),
                        "%",
                      )}
                      <div className="opla-analytics_legend">Errors rate</div>
                    </div>
                  </div>
                  <div className="opla-analytics_metric">
                    <div>
                      {this.renderingValue(
                        metrics.responses.speed.toFixed(2),
                        "ms",
                      )}
                      <div className="opla-analytics_legend">Response time</div>
                    </div>
                  </div>
                </div>
              </Panel>
            </Cell>
          </Inner>
        </Grid>
      </div>
    );
  }
}

AnalyticsBase.defaultProps = {
  isLoading: false,
  isSignedIn: false,
  metrics: null,
};

AnalyticsBase.propTypes = {
  isLoading: PropTypes.bool,
  isSignedIn: PropTypes.bool,
  metrics: PropTypes.shape({}),
  selectedBotId: PropTypes.string,
  fetchMetrics: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { metrics, user } = state;

  const isSignedIn = user ? user.isSignedIn : false;

  return {
    metrics: metrics.metrics,
    isLoading: metrics.loading,
    selectedBotId: state.app.selectedBotId,
    isSignedIn,
  };
};

const mapDispatchToProps = (dispatch) => ({
  fetchMetrics: (botId) => dispatch(apiGetMetricsRequest(botId)),
});

// prettier-ignore
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AnalyticsBase);
