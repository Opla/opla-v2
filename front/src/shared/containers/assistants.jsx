/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router";
import Panel from "zoapp-front/dist/components/panel";
import { Grid, Inner } from "zrmc";
import AssistantCard from "../components/assistantCard";

export class AssistantsBase extends Component {
  onSelect = (index) => {
    if (index > -1) {
      // TODO select new bot
      this.props.history.push("/factory");
    } else {
      this.props.history.push("/create");
    }
  };

  render() {
    const { bots, multiBots, selectedBotIndex } = this.props;

    let create;
    if (multiBots || bots.length < 1) {
      create = (
        <AssistantCard
          item={{
            name: "Create a new assistant",
            icon: (
              <svg viewBox="0 0 24 24">
                <path d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z" />
              </svg>
            ),
          }}
          onSelect={this.onSelect}
        />
      );
    }
    return (
      <div className="opla-assistants zui-layout__content zui-color--white">
        <Panel title="Assistants">
          <Grid>
            <Inner>
              {bots.map((bot, index) => (
                <AssistantCard
                  key={bot.id}
                  item={bot}
                  index={index}
                  selectedIndex={selectedBotIndex}
                  onSelect={this.onSelect}
                />
              ))}
              {create}
            </Inner>
          </Grid>
        </Panel>
      </div>
    );
  }
}

AssistantsBase.propTypes = {
  isSignedIn: PropTypes.bool.isRequired,
  bots: PropTypes.arrayOf(PropTypes.shape({})),
  multiBots: PropTypes.bool,
  selectedBotIndex: PropTypes.number,
  history: PropTypes.shape({ length: PropTypes.number, push: PropTypes.func })
    .isRequired,
};

const mapStateToProps = (state) => {
  const { admin } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const selectedBotId = state.app ? state.app.selectedBotId : null;
  const selectedBotIndex = selectedBotId
    ? state.app.project.selectedIndex
    : null;
  const bots = admin ? admin.bots : [];
  const multiBots = admin ? admin.params.multiProjects : false;
  return { isSignedIn, bots, multiBots, selectedBotIndex };
};

export default withRouter(connect(mapStateToProps)(AssistantsBase));
