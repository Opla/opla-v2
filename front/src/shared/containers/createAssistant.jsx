/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { Component } from "react";
import PropTypes from "prop-types";
import Zrmc, { Select, MenuItem, Button, TextField } from "zrmc";
import { connect } from "react-redux";
import { withRouter } from "react-router";
import ProcessingDialog from "zoapp-front/dist/containers/processingDialog";
import { appSetTitleName, setMessage } from "zoapp-front/dist/actions/app";

import TemplatesList from "../components/templatesList";
import {
  apiCreateBot,
  apiGetTemplatesRequest,
  apiGetLanguagesRequest,
} from "../actions/api";

const boxStyle = {
  margin: "16px",
};

const headerStyle = {
  padding: "16px",
};

const h4 = {
  marginTop: "0px",
  marginBottom: "16px",
  fontSize: "16px",
  color: "rgba(0, 0, 0, 0.87)",
};
const secText = {
  color: "rgba(0, 0, 0, 0.54)",
};

export class CreateAssistantBase extends Component {
  constructor(props) {
    super();
    this.state = {
      name: "",
      language: null,
      loading: false,
      selectedTemplate: null,
      template: null,
    };
    props.appSetTitleName("Create an assistant", "Create");
  }

  componentDidMount() {
    this.props.apiGetTemplates();
    this.props.apiGetLanguages();
  }

  componentDidUpdate(prevProps) {
    if (
      prevProps.templates !== this.props.templates &&
      this.state.selectedTemplate === null
    ) {
      const selectedTemplate = this.props.templates.findIndex(
        (template) => template.name === "Empty",
      );

      this.onSelectTemplate(
        selectedTemplate,
        this.props.templates[selectedTemplate],
      );
    }
    if (this.state.loading && this.props.isLoading === false) {
      this.handleCloseCreateDialog();
    }
  }

  onImportTemplate = (data) => {
    try {
      const json = JSON.parse(data);
      this.onSelectTemplate(3, json);
    } catch (e) {
      this.props.setMessage("imported template is not a valid JSON document");
    }
  };

  onSelectTemplate = (selected, data) => {
    let template = data;
    if (!template) {
      template = this.props.templates[selected];
    }
    this.setState({ selectedTemplate: selected, template });
  };

  handleCloseCreateDialog = () => {
    this.setState({ loading: false });
    Zrmc.closeDialog();
    if (!this.props.error) {
      // TODO select Bot created
      this.props.history.push("/factory");
    }
  };

  handleCreate = (e) => {
    e.preventDefault();

    const { name, email, language, template, loading } = this.state;

    if (template === null) {
      this.props.setMessage("please select a template");
      return;
    }

    if (loading === false) {
      const botParams = {
        name,
        email,
        template,
        language,
      };

      this.setState({ loading: true });

      Zrmc.showDialog(
        <ProcessingDialog open onClosed={this.handleCloseDialog} />,
      );

      this.props.createBot(botParams);
    } else {
      // TODO display errors in dialogs
    }
  };

  handleLanguageChange = (language) => {
    this.setState({ language });
  };

  createChangeHandler = (field) => (e) => {
    this.setState({ [field]: e.target.value });
  };

  render() {
    const { name, email, selectedTemplate: selected } = this.state;

    let selectedLanguageIndex = 0;
    const languagesItems = this.props.languages.map((language, index) => {
      if (language.default === true) {
        selectedLanguageIndex = index;
      }
      return (
        <MenuItem
          key={language.id}
          selected={language.default}
          value={language.id}
        >
          {language.name}
        </MenuItem>
      );
    });

    // TODO json only for instance
    const acceptImport = "application/json";
    return (
      <div className="zui-layout__content zui-color--grey-100">
        <section className="zui-color--white zui-shadow--2dp" style={boxStyle}>
          <div style={headerStyle}>
            <h4 style={h4}>Welcome</h4>
            <div style={secText}>
              Create your open conversational robot at the speed of light. Just
              follow the instructions, no coding required !
            </div>
          </div>
        </section>
        <section className="zui-color--white zui-shadow--2dp" style={boxStyle}>
          <div style={headerStyle}>
            <h4 style={h4}>Templates</h4>
            <div style={secText}>
              Choose a prebuild asssistant, import one or select an empty model.
            </div>
          </div>
          {this.props.templates &&
            this.props.templates.length > 0 && (
              <TemplatesList
                items={this.props.templates}
                selectedItem={selected}
                onSelect={this.onSelectTemplate}
                onImport={this.onImportTemplate}
                acceptImport={acceptImport}
              />
            )}
        </section>
        <form
          id="create-assistant-form"
          autoComplete="new-password"
          onSubmit={this.handleCreate}
        >
          <section
            className="zui-color--white zui-shadow--2dp"
            style={boxStyle}
          >
            <div style={headerStyle}>
              <h4 style={h4}>Informations needed</h4>
              <div style={secText}>
                Please provide the following information. Don&apos;t worry. You
                can always change them later.
              </div>
              <div>
                <TextField
                  id="create-assistant-name"
                  onChange={this.createChangeHandler("name")}
                  defaultValue={name}
                  label="Assistant name"
                  style={{ width: "400px" }}
                  required
                />
              </div>
              <div>
                <TextField
                  id="create-assistant-email"
                  type="email"
                  onChange={this.createChangeHandler("email")}
                  defaultValue={email}
                  label="Contact email"
                  style={{ width: "400px" }}
                  required
                />
              </div>
              <div>
                <Select
                  id="create-assistant-language"
                  label="Choose language"
                  onSelected={this.handleLanguageChange}
                  style={{ width: "400px" }}
                  selectedIndex={selectedLanguageIndex}
                  required
                >
                  {languagesItems}
                </Select>
              </div>
            </div>
          </section>
          <section style={boxStyle}>
            <div style={headerStyle}>
              <Button type="submit" raised>
                Create
              </Button>
            </div>
          </section>
        </form>
      </div>
    );
  }
}

CreateAssistantBase.defaultProps = {
  error: null,
};

CreateAssistantBase.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  isSignedIn: PropTypes.bool.isRequired,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.shape({})]),
  createBot: PropTypes.func.isRequired,
  appSetTitleName: PropTypes.func.isRequired,
  apiGetTemplates: PropTypes.func.isRequired,
  apiGetLanguages: PropTypes.func.isRequired,
  setMessage: PropTypes.func.isRequired,
  history: PropTypes.shape({ length: PropTypes.number, push: PropTypes.func })
    .isRequired,
  templates: PropTypes.array,
  languages: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
  const { admin, error, templates, languages } = state.app;
  const isSignedIn = state.user ? state.user.isSignedIn : false;
  const isLoading = state.app.loading || false;
  return {
    admin,
    isLoading,
    isSignedIn,
    error,
    templates,
    languages,
  };
};

const mapDispatchToProps = (dispatch) => ({
  createBot: (botParams) => {
    dispatch(apiCreateBot(botParams));
  },
  appSetTitleName: (titleName) => {
    dispatch(appSetTitleName(titleName));
  },
  setMessage: (message) => {
    dispatch(setMessage(message));
  },
  apiGetTemplates: () => {
    dispatch(apiGetTemplatesRequest());
  },
  apiGetLanguages: () => {
    dispatch(apiGetLanguagesRequest());
  },
});

// prettier-ignore
export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  )(CreateAssistantBase),
);
