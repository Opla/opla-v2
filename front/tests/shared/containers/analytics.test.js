/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import { AnalyticsBase } from "shared/containers/analytics";

describe("containers/Analytics", () => {
  const defaultMetrics = {
    users: {
      count: 123,
    },
    conversations: {
      count: 1200,
      messages_per_conversation: 4,
    },
    sessions: {
      duration: 5000,
    },
    errors: {
      rate: 0.230115,
    },
    responses: {
      speed: 40.219032,
    },
  };
  it("renders and update correctly", () => {
    const fetchMetricsSpy = jest.fn();

    const component = renderer.create(
      <AnalyticsBase
        fetchMetrics={fetchMetricsSpy}
        isSignedIn
        metrics={defaultMetrics}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(fetchMetricsSpy).not.toHaveBeenCalled();

    component.update(
      <AnalyticsBase
        fetchMetrics={fetchMetricsSpy}
        isSignedIn
        renderingValue
        selectedBotId={"bot1"}
        metrics={defaultMetrics}
      />,
    );
    expect(fetchMetricsSpy).toHaveBeenCalledWith("bot1");
  });

  it("renders a Loading component when user is not signed in", () => {
    const component = renderer.create(
      <AnalyticsBase fetchMetrics={jest.fn()} isSignedIn={false} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a Loading component when isLoading is true", () => {
    const component = renderer.create(
      <AnalyticsBase fetchMetrics={jest.fn()} isSignedIn isLoading />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a Loading component when no metrics", () => {
    const metrics = null;

    const component = renderer.create(
      <AnalyticsBase
        fetchMetrics={jest.fn()}
        isSignedIn
        renderingValue
        metrics={metrics}
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("fetch metrics when selected bot change", () => {
    const fetchMetricsSpy = jest.fn();

    const wrapper = shallow(
      <AnalyticsBase
        fetchMetrics={fetchMetricsSpy}
        isSignedIn
        metrics={defaultMetrics}
      />,
    );
    wrapper.setProps({ selectedBotId: "abc" });
    expect(fetchMetricsSpy).toHaveBeenCalledWith("abc");
  });
});
