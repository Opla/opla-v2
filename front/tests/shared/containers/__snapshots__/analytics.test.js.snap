// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Analytics renders a Loading component when isLoading is true 1`] = `
<div
  className="mdc-card"
  style={
    Object {
      "margin": "auto",
      "width": "512px",
    }
  }
>
  <section
    className="mdc-card__primary"
  >
    
    <h1
      className="mdc-card__title"
    >
      Processing
    </h1>
    
  </section>
  <section
    className="mdc-card__supporting-text"
  >
    Please wait ...
  </section>
  <section
    className="mdc-card__actions"
  />
</div>
`;

exports[`containers/Analytics renders a Loading component when no metrics 1`] = `
<div
  className="mdc-card"
  style={
    Object {
      "margin": "auto",
      "width": "512px",
    }
  }
>
  <section
    className="mdc-card__primary"
  >
    
    <h1
      className="mdc-card__title"
    >
      Processing
    </h1>
    
  </section>
  <section
    className="mdc-card__supporting-text"
  >
    Please wait ...
  </section>
  <section
    className="mdc-card__actions"
  />
</div>
`;

exports[`containers/Analytics renders a Loading component when user is not signed in 1`] = `
<div>
  You need to sign in...
</div>
`;

exports[`containers/Analytics renders and update correctly 1`] = `
<div
  className="opla-analytics zui-layout__content zui-color--white"
>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="zui-color--white mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
      >
        <div
          className="zap-panel"
        >
          <div
            className="zap-panel_header"
          >
            <div
              className="zap-panel_title"
            >
              
              <div>
                Key metrics
              </div>
            </div>
            
          </div>
          <div
            className="zap-panel_desc"
          />
          <div
            className="opla-analytics_subpanel"
          >
            <div
              className="opla-analytics_metric"
            >
              <div>
                <span
                  className="opla-analytics_value"
                >
                  123
                  <span
                    className="opla-analytics_unit"
                  />
                </span>
                <div
                  className="opla-analytics_legend"
                >
                  Users
                </div>
              </div>
            </div>
            <div
              className="opla-analytics_metric"
            >
              <div>
                <span
                  className="opla-analytics_value"
                >
                  1200
                  <span
                    className="opla-analytics_unit"
                  />
                </span>
                <div
                  className="opla-analytics_legend"
                >
                  Conversations
                </div>
              </div>
            </div>
            <div
              className="opla-analytics_metric"
            >
              <div>
                <span
                  className="opla-analytics_value"
                >
                  4
                  <span
                    className="opla-analytics_unit"
                  />
                </span>
                <div
                  className="opla-analytics_legend"
                >
                  Messages
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="zui-color--white mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
        style={
          Object {
            "paddingTop": "96px",
          }
        }
      >
        <div
          className="zap-panel"
        >
          <div
            className="zap-panel_header"
          >
            <div
              className="zap-panel_title"
            >
              
              <div>
                Platform metrics
              </div>
            </div>
            
          </div>
          <div
            className="zap-panel_desc"
          />
          <div
            className="opla-analytics_subpanel"
          >
            <div
              className="opla-analytics_metric"
            >
              <div>
                <span
                  className="opla-analytics_value"
                >
                  5000
                  <span
                    className="opla-analytics_unit"
                  >
                    ms
                  </span>
                </span>
                <div
                  className="opla-analytics_legend"
                >
                  Session duration
                </div>
              </div>
            </div>
            <div
              className="opla-analytics_metric"
            >
              <div>
                <span
                  className="opla-analytics_value"
                >
                  23.01
                  <span
                    className="opla-analytics_unit"
                  >
                    %
                  </span>
                </span>
                <div
                  className="opla-analytics_legend"
                >
                  Errors rate
                </div>
              </div>
            </div>
            <div
              className="opla-analytics_metric"
            >
              <div>
                <span
                  className="opla-analytics_value"
                >
                  40.22
                  <span
                    className="opla-analytics_unit"
                  >
                    ms
                  </span>
                </span>
                <div
                  className="opla-analytics_legend"
                >
                  Response time
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
